spec/beaker/testmode_switcher/dsl_spec.rb

```diff
@@ -3,39 +3,42 @@
 require 'beaker/testmode_switcher/dsl'
 
 describe Beaker::TestmodeSwitcher::DSL do
+  let(:hosts) {}
+  let(:logger) {}
+
   describe '#create_remote_file_ex' do
     it 'is callable' do
-      expect(Beaker::TestmodeSwitcher.runner).to receive(:create_remote_file_ex)
+      expect(Beaker::TestmodeSwitcher.runner(hosts, logger)).to receive(:create_remote_file_ex)
       create_remote_file_ex('/tmp/foo', 'content', {})
     end
   end
   describe '#scp_to_ex' do
     it 'is callable' do
-      expect(Beaker::TestmodeSwitcher.runner).to receive(:scp_to_ex)
+      expect(Beaker::TestmodeSwitcher.runner(hosts, logger)).to receive(:scp_to_ex)
       scp_to_ex('/tmp/from', '/tmp/to')
     end
   end
   describe '#shell_ex' do
     it 'is callable' do
-      expect(Beaker::TestmodeSwitcher.runner).to receive(:shell_ex)
+      expect(Beaker::TestmodeSwitcher.runner(hosts, logger)).to receive(:shell_ex)
       shell_ex('cmd', {})
     end
   end
   describe '#resource' do
     it 'is callable' do
-      expect(Beaker::TestmodeSwitcher.runner).to receive(:resource)
+      expect(Beaker::TestmodeSwitcher.runner(hosts, logger)).to receive(:resource)
       resource('type', 'name', {})
     end
   end
   describe '#execute_manifest' do
     it 'is callable' do
-      expect(Beaker::TestmodeSwitcher.runner).to receive(:execute_manifest)
+      expect(Beaker::TestmodeSwitcher.runner(hosts, logger)).to receive(:execute_manifest)
       execute_manifest('manifest', {})
     end
   end
 
   describe '#runner' do
-    let(:subject) { Beaker::TestmodeSwitcher.runner }
+    let(:subject) { Beaker::TestmodeSwitcher.runner(hosts, logger) }
```

Comment:
Looking at all that code duplication, it would make sense to pull this `let` out into the top-level block. Also there is `subject{}`, then you can use `is_expected.to` instead of `expect(subject).to`